### DISCUSSION TITLE
Welcome to buildroot_bouffalo Discussions!

### DISCUSSION THREAD

**Discussion by Fishwaldo** [openbouffalo/buildroot_bouffalo#30] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Hunter1753** [comment]
Hi, I'm Alex and I would like to start developing one of the missing drivers for the bl808.

What's currently holding me back is that I don't understand how I can do kernel development for this, as I can't even get buildroot to use the kernel directory I specify.

I would appreciate any help with getting started :)

**Comment by Fishwaldo** [comment]
Its best to develop outside of buildroot for kernel work. The main focus of buildroot is the rootfs.

To do kernel development, basically setup your own kernel tree (we are working from https://github.com/arm000/linux-bl808/tree/linux-next/mboxic mainly) and build the kernel there. 

Then to flash, copy fw_jump.bin and merge.py from the buildroot and basically copy these steps:
https://github.com/openbouffalo/buildroot_bouffalo/blob/main/board/pine64/ox64/post-build.sh (minus the "create filesystem image")

then burn the resultant bin file using the IOT tab as per the instructions

**Reply by Hunter1753** [reply]
So I just install the compilers and I then make the kernel with
```make ARCH=riscv CROSS_COMPILE=/usr/bin/riscv64-unknown-elf-```
right?

I will try this out tonight, thank you!

**Reply by Hunter1753** [reply]
ok, it worked with the SDK, thanks for your help
